src/middleware.js

```diff
@@ -52,6 +53,22 @@ export function withCarHandler (handler) {
   }
 }
 
+/**
+ * Middleware that will serve blobs if `/blob/:multihash` is found in the path.
+ * If path does not start with `/blob/` it delegates to the next middleware.
+ *
+ * @type {import('@web3-storage/gateway-lib').Middleware<import('@web3-storage/gateway-lib').Context, import('@web3-storage/gateway-lib').Context, Environment>}
+ */
+export function withBlobHandler (handler) {
+  return async (request, env, ctx) => {
+    const url = new URL(request.url)
+    if (!url.pathname.startsWith('/blob/')) {
```

Comment:
Do we even need the `/blob/` interface as in is there reason why we would not use RAW cid handler instead ? If we did query content claims for raw blocks we would effectively get pointer to the file covering it's full content would not we ?

Reply:
Agree, lets move this into the raw handler.

Reply:
yes, I also agree, and think actually having `/blob` will just make everything more complicated on the reads side